src/preloader/TcpSocketElectron.ts

```diff
@@ -14,55 +14,95 @@ export class TcpSocketElectron {
   readonly id: number;
   readonly host: string;
   readonly port: number;
-  private _socket: net.Socket;
-  private _messagePort: MessagePort;
-  private _api = new Map<string, RpcHandler>([
-    ["remoteAddress", (callId) => this._apiResponse(callId, this.remoteAddress())],
-    ["localAddress", (callId) => this._apiResponse(callId, this.localAddress())],
-    ["fd", (callId) => this._apiResponse(callId, this.fd())],
+  #socket: net.Socket;
+  #messagePort: MessagePort;
+  #api = new Map<string, RpcHandler>([
+    [
+      "remoteAddress",
+      (callId) => {
+        this.#apiResponse(callId, this.remoteAddress());
+      },
+    ],
+    [
+      "localAddress",
+      (callId) => {
+        this.#apiResponse(callId, this.localAddress());
+      },
+    ],
+    [
+      "fd",
+      (callId) => {
+        this.#apiResponse(callId, this.fd());
+      },
+    ],
     [
       "setKeepAlive",
       (callId, args) => {
         const enable = args[0] as boolean | undefined;
         const initialDelay = args[1] as number | undefined;
         this.setKeepAlive(enable, initialDelay);
-        this._apiResponse(callId);
+        this.#apiResponse(callId);
       },
     ],
     [
       "setTimeout",
       (callId, args) => {
         const timeout = args[0] as number;
         this.setTimeout(timeout);
-        this._apiResponse(callId);
+        this.#apiResponse(callId);
       },
     ],
     [
       "setNoDelay",
       (callId, args) => {
         const noDelay = args[0] as boolean | undefined;
         this.setNoDelay(noDelay);
-        this._apiResponse(callId);
+        this.#apiResponse(callId);
+      },
+    ],
+    [
+      "connected",
+      (callId) => {
+        this.#apiResponse(callId, this.connected());
       },
     ],
-    ["connected", (callId) => this._apiResponse(callId, this.connected())],
     [
       "connect",
       (callId, _) => {
         this.connect()
-          .then(() => this._apiResponse(callId, undefined))
-          .catch((err: Error) => this._apiResponse(callId, String(err.stack ?? err)));
+          .then(() => {
+            this.#apiResponse(callId, undefined);
+          })
+          .catch((err: Error) => {
+            this.#apiResponse(callId, String(err.stack ?? err));
+          });
+      },
+    ],
+    [
+      "close",
+      (callId) => {
+        this.close();
+        this.#apiResponse(callId);
+      },
+    ],
+    [
+      "dispose",
+      (callId) => {
+        this.dispose();
+        this.#apiResponse(callId);
```

Comment:
back to the original behaviour, passing `this.dipose()` via params.

```suggestion
        this.#apiResponse(callId, this.dispose());
```

Reply:
Done